### PR TITLE
MVT Provider refactoring

### PR DESCRIPTION
**Pull request by doublebyte1**
# Overview
This PR introduces a refactoring in the MVT provider class. The MVT Provider becomes a base class, which is implemented through different providers. The tippecanoe and elastic MVT providers are also offered in this PR.

Tests, documentation and docker examples are also updated accordingly.

# Related Issue / Discussion
https://github.com/geopython/pygeoapi/issues/1450

# Contributions and Licensing

(as per https://github.com/geopython/pygeoapi/blob/master/CONTRIBUTING.md#contributions-and-licensing)

- [x] I'd like to contribute [feature X|bugfix Y|docs|something else] to pygeoapi. I confirm that my contributions to pygeoapi will be compatible with the pygeoapi license guidelines at the time of contribution.
- [x] I have already previously agreed to the pygeoapi Contributions and Licensing Guidelines

### CODE REVIEW

File: docker/examples/MVT-elastic/ES/Dockerfile
Comment (tomkralidis) [inline docker/examples/MVT-elastic/ES/Dockerfile:1]: Rename to `docker/examples/mvt-elastic/ES/Dockerfile`

File: docker/examples/MVT-tippecanoe/README.md
Comment (tomkralidis) [inline docker/examples/MVT-tippecanoe/README.md:1]: Rename to `docker/examples/mvt-tippecanoe/README.md`

File: docs/source/data-publishing/ogcapi-tiles.rst [hunk 1]
```diff
@@ -7,50 +7,42 @@ Publishing tiles to OGC API - Tiles
...
-The MVT provider plugin provides access to `Mapbox Vector Tiles`_.
-
-Remote data sources can be any external service (i.e. Elasticsearch), by providing a URL
-template.
-
-.. note::
-   Currently, the URL templating in this provider supports the following formats: `/{z}/{x}/{y}` or `/{z}/{y}/{x}`.
-   For additional formats: feel free to file an `issue <https://github.com/geopython/pygeoapi/issues>`_.
+MVT-tippecanoe
+^^^^^^^^^^^^^^^
```
Comment (tomkralidis) [inline docs/source/data-publishing/ogcapi-tiles.rst:33]: Align

File: docs/source/data-publishing/ogcapi-tiles.rst [hunk 2]
```diff
@@ -7,50 +7,42 @@ Publishing tiles to OGC API - Tiles
...
 
+This provider gives support to serving tiles generated using `Mapbox Tippecanoe <https://github.com/mapbox/tippecanoe>`_. 
+The tiles can be integrated from a path on disk, or from a static url (e.g.: from an S3 or MinIO bucket). 
+In both cases, they have to be rendered before using pygeoapi.
 
-This code block shows how to configure pygeoapi to read Mapbox vector tiles, from disk or a URL.
+This code block shows how to configure pygeoapi to read Mabox vector tiles generated with tippecanoe, from disk or a URL.
```
Comment (tomkralidis) [inline docs/source/data-publishing/ogcapi-tiles.rst:39]: `s/Mabox/Mapbox/`

File: pygeoapi/plugin.py [hunk 3]
```diff
@@ -50,7 +50,10 @@
         'Hateoas': 'pygeoapi.provider.hateoas.HateoasProvider',
         'MapScript': 'pygeoapi.provider.mapscript_.MapScriptProvider',
         'MongoDB': 'pygeoapi.provider.mongo.MongoProvider',
-        'MVT': 'pygeoapi.provider.mvt.MVTProvider',
+        'MVT-tippecanoe':
```
Comment (tomkralidis) [inline pygeoapi/plugin.py:53]: Put these on single lines and `#noqa`

**Review by tomkralidis — COMMENTED** [review]
Minor change requests -- great work!

File: .gitignore [hunk 4]
```diff
@@ -125,3 +125,7 @@ pygeoapi/db.sqlite3
 
 # Pycharm project files
 .idea
+
+# ES data folder
+.pygeoapi/docker/examples/elastic/ES/data
+.pygeoapi/docker/examples/MVT-elastic/ES/data
```
Comment (tomkralidis) [inline .gitignore:131]: `s/MVT-elastic/mvt-elastic/`

**Review by tomkralidis — CHANGES_REQUESTED** [review]

**Review by tomkralidis — APPROVED** [review]

**Comment by tomkralidis** [comment]
Over to @francbartoli for review/approve.

File: pygeoapi/provider/mvt_elastic.py [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+        Initialize object
+
+        :param provider_def: provider definition
+
+        :returns: pygeoapi.provider.MVT.MVTElasticProvider
+        """
+
+        super().__init__(BaseMVTProvider)
+
+        self.tile_type = 'vector'
```
Comment (francbartoli) [inline pygeoapi/provider/mvt_elastic.py:60]: This is redundant because already set in the super class

**Review by francbartoli — CHANGES_REQUESTED** [review]

**Review by francbartoli — APPROVED** [review]
Great work here @doublebyte1

**Comment by justb4** [comment]
Shortly after this merge we got a [GeoHealthCheck failure report](https://demo.geohealthcheck.org/resource/575) for the demo server. Maybe good to always check after a merge here as the pygeoapi Docker image is deployed there at https://demo.pygeoapi.io/master via CI/CD-chain (may take some time though). But looks related, maybe a config thing, to this PR.

```
pygeoapi.plugin.InvalidPluginError: Plugin MVT not found
ERROR: openapi.yml could not be generated ERROR
```

**Comment by tomkralidis** [comment]
Thanks @justb4; PR in https://github.com/geopython/demo.pygeoapi.io/pull/48

Note: also see PR in #1463 that adds a checkbox to PR templates.

**Pull request merged**